In [1]:
import pandas as pd

In [2]:
raw_df = pd.read_csv("bops_bm.csv")
raw_df 

id (store)  year  month  week  usa  after   sales 
0              1  2011      4    16    0      0  118,691
1              1  2011      4    17    0      0  113,804
2              1  2011      4    18    0      0  172,104
3              1  2011      5    19    0      0  105,591
4              1  2011      5    20    0      0   94,884
...          ...   ...    ...   ...  ...    ...      ...
4531         169  2012      3    12    1      1   56,331
4532         169  2012      3    13    1      1   74,388
4533         169  2012      3    14    1      1   67,671
4534         169  2012      4    15    1      1   50,140
4535         169  2012      4    16    1      1   75,023

[4536 rows x 7 columns]

In [3]:
df = raw_df.copy().replace(',', '', regex=True)

###  B&M sales Analysis

In [4]:
before_bops_df = df[df['after'] == 0].reset_index(drop = True)
after_bops_df = df[df['after'] == 1].reset_index(drop = True)

La cantidad de dinero generado por ventas antes de BOPS en B&M fue:

In [5]:
len(before_bops_df)*67646

153421128

153,421,128 $

La cantidad de dinero generado por ventas despues de BOPS en B&M fue:

In [6]:
len(after_bops_df)*60101

136309068

136,309,068 $

Esto consiste en una pérdida de:

In [7]:
loss_BM = len(before_bops_df)*67646-len(after_bops_df)*60101
loss_BM 

17112060

17,112,060 $

### Online sales analysis

In [8]:
raw_df2 = pd.read_csv("bops_online.csv")
# raw_df2 = raw_df2[['id (DMA)','year','month','week','after','close', 'sales']]
df2 = raw_df2.iloc[:,0:7].copy().replace(',', '', regex=True)
df2

id (DMA)  year  month  week  after  close sales 
0             1  2011      4    17      0      1  18564
1             1  2011      4    18      0      1  30883
2             1  2011      5    19      0      1  37425
3             1  2011      5    20      0      1  32563
4             1  2011      5    21      0      1  35773
...         ...   ...    ...   ...    ...    ...    ...
10705       210  2012      2    10      1      0   9357
10706       210  2012      3    11      1      0   3013
10707       210  2012      3    12      1      0   3155
10708       210  2012      3    13      1      0   2925
10709       210  2012      3    14      1      0   3701

[10710 rows x 7 columns]

In [9]:
before_bops_df2 = df2[df2['after'] == 0].reset_index(drop = True)
after_bops_df2 = df2[df2['after'] == 1].reset_index(drop = True)

La cantidad de dinero generado en ventas online antes de BOPS fue:

14,738 $ es el promedio generado por ventas antes de desplegar BOPS

In [10]:
len(before_bops_df2)*14738

80469480

80,469,480 $

12,734 $ es el promedio generado por ventas despues de desplegar BOPS

La cantidad de dinero generado por ventas online despues de BOPS fue:

In [11]:
len(after_bops_df2)*12734

66853500

66,853,500 $

Esto consiste en una pérdida de:

In [35]:
loss_online = len(before_bops_df2)*14738-len(after_bops_df2)*12734
loss_online

13615980

13,615,980 $

Ahora sumamos las pérdidas online más las pérdidas en B&M para saber cuanto dinero se perdió a partir del programa.

In [36]:
perdidas_totales = loss_online+ loss_BM
perdidas_totales

30728040

### 2. ¿Cuántos millones de dólares se ganaron o perdieron a partir del programa? Explica tu razonamiento y metodología.

En base a la metodología empleada en las celdas de arriba, a partir del programa se perdieron 30.728040 millones de dolares.

### Expansion a Canada

In [14]:
canada_df = df[df['usa'] == 0]
canada_df_before = canada_df[canada_df['after'] == 0]
canada_df_after = canada_df[canada_df['after'] == 1]

In [15]:
# canada_df_before.iloc[:,6] = canada_df_before.iloc[:,6].replace(',', '', regex=True))
# canada_df_after.iloc[:,6] = canada_df_after.iloc[:,6].replace(',', '', regex=True))

In [16]:
canada_df_before['sales'] = pd.to_numeric(canada_df_before.iloc[:,6],errors='coerce')
canada_df_after['sales'] = pd.to_numeric(canada_df_after.iloc[:,6],errors='coerce')
# df['DataFrame Column'] = pd.to_numeric(df['DataFrame Column'],errors='coerce')

<ipython-input-16-f1340f394688>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  canada_df_before['sales'] = pd.to_numeric(canada_df_before.iloc[:,6],errors='coerce')
<ipython-input-16-f1340f394688>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  canada_df_after['sales'] = pd.to_numeric(canada_df_after.iloc[:,6],errors='coerce')


### Abajo se puede ver el promedio de las ventas antes y despues de que se implementara BOPS

In [17]:
canada_df_before['sales'].mean()

66862.25925925926

In [18]:
canada_df_after['sales'].mean()

56325.23965141612

En Canadá se redujeron las ventas por este porcentaje:

In [19]:
(canada_df_before['sales'].mean()-canada_df_after['sales'].mean())/canada_df_before['sales'].mean()

0.15759293395973514

In [20]:
usa_df = df[df['usa'] == 1]
usa_df_before = usa_df[usa_df['after'] == 0]
usa_df_after = usa_df[usa_df['after'] == 1]

In [21]:
usa_df_before['sales'] = pd.to_numeric(usa_df_before.iloc[:,6],errors='coerce')
usa_df_after['sales'] = pd.to_numeric(usa_df_after.iloc[:,6],errors='coerce')

<ipython-input-21-d601a8a2d055>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df_before['sales'] = pd.to_numeric(usa_df_before.iloc[:,6],errors='coerce')
<ipython-input-21-d601a8a2d055>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df_after['sales'] = pd.to_numeric(usa_df_after.iloc[:,6],errors='coerce')


In [22]:
usa_df_before['sales'].mean()

67844.49695964622

In [23]:
usa_df_after['sales'].mean()

61058.9325594251

En estados unidos se redujeron las ventas por este porcentage:

In [24]:
(usa_df_before['sales'].mean()-usa_df_after['sales'].mean())/usa_df_before['sales'].mean()

0.1000164302825793

### Online Sales analysis

In [29]:
far_df = df2[df2['close'] == 0]
far_df_before = far_df[far_df['after'] == 0]
far_df_after = far_df[far_df['after'] == 1]

In [30]:
far_df_before['sales'] = pd.to_numeric(far_df_before.iloc[:,6],errors='coerce')
far_df_after['sales'] = pd.to_numeric(far_df_after.iloc[:,6],errors='coerce')

<ipython-input-30-0c018d99226b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  far_df_before['sales'] = pd.to_numeric(far_df_before.iloc[:,6],errors='coerce')
<ipython-input-30-0c018d99226b>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  far_df_after['sales'] = pd.to_numeric(far_df_after.iloc[:,6],errors='coerce')


In [31]:
(far_df_before['sales'].mean()-far_df_after['sales'].mean())/far_df_before['sales'].mean()

0.12057864485743752

In [32]:
close_df = df2[df2['close'] == 1]
close_df_before = close_df[close_df['after'] == 0]
close_df_after = close_df[close_df['after'] == 1]

In [33]:
close_df_before['sales'] = pd.to_numeric(close_df_before.iloc[:,6],errors='coerce')
close_df_after['sales'] = pd.to_numeric(close_df_after.iloc[:,6],errors='coerce')

<ipython-input-33-3427d153369b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  close_df_before['sales'] = pd.to_numeric(close_df_before.iloc[:,6],errors='coerce')
<ipython-input-33-3427d153369b>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  close_df_after['sales'] = pd.to_numeric(close_df_after.iloc[:,6],errors='coerce')


In [34]:
(close_df_before['sales'].mean()-close_df_after['sales'].mean())/close_df_before['sales'].mean()

0.15484335054502593

### 1. ¿Deberían expandirse a Canadá?

No deberían de expandirse a Canada porque BOPS no es una estrategía que le va a generar crecimiento en ventas a Home & Kitchen, puesto que desde que se comenzó a emplear no lo ha hecho. A pesar de que en Canadá hubo un mayor porcentage de decrecimiento en ventas comparado con USA, esto no es suficiente razón como para que sea alentador la idea de expandirse a Canadá. Sería más favorable mejorar los productos de Home & Kitchen e identificar otras estrategías para generar el crecimiento deseado. 